### PR TITLE
Feature/integrate notifications

### PR DESCRIPTION
**Pull request by maismin**
- [x] Notifications are pulled from the database when a new product is added or a lower price is found

- [x] Dismiss notifications

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -38,10 +38,15 @@ const styles = () => ({
   },
 });
 
-function App({ loadUser, loading }) {
+function App({ loadUser, loadNotifications, loading }) {
   useEffect(() => {
-    loadUser();
-  }, [loadUser]);
+    async function load() {
```
Comment (Esodey) [inline client/src/App.js:43]: custom hook right?

File: client/src/components/Dashboard/ShoppingLists.js [hunk 2]
```diff
@@ -72,29 +70,17 @@ function mapListsToCards(shoppingLists, classes, handleOpenList) {
...
-    setOpenAddItem(true);
-  };
-
-  const handleCloseAddItem = () => {
-    setOpenAddItem(false);
-  };
-
-  // eslint-disable-next-line no-unused-vars, no-shadow
-  const { getLists, clearErrors, error, lists, loading } = props;
+  const {
```
Comment (Esodey) [inline client/src/components/Dashboard/ShoppingLists.js:74]: Lot of variables. Makes sense

File: client/src/pages/Dashboard.js [hunk 3]
```diff
@@ -14,12 +16,43 @@ const useStyles = makeStyles({
 
 export default function Dashboard() {
   const classes = useStyles();
+  const [openList, setOpenList] = useState(false);
+  const [openAddItem, setOpenAddItem] = useState(false);
+  const [listToDisplay, setListToDisplay] = useState(null);
```
Comment (Esodey) [inline client/src/pages/Dashboard.js:21]: Good renaming

File: client/src/components/SignIn/SignInForm.js [hunk 4]
```diff
@@ -57,14 +59,15 @@ function SignInForm({
     return <Redirect to={from.pathname} />;
   }
 
-  const handleSubmit = event => {
+  const handleSubmit = async event => {
```
Comment (Esodey) [inline client/src/components/SignIn/SignInForm.js:62]: async makes complete sense

File: client/src/components/SocketClient.js [hunk 5]
```diff
@@ -18,20 +21,24 @@ function SocketClient({ isAuthenticated }) {
     socket.emit('authenticate');
```
Comment (Esodey) [inline client/src/components/SocketClient.js:22]: Still need to look into how these sockets work

File: client/src/pages/Dashboard.js [hunk 6]
```diff
@@ -14,12 +16,43 @@ const useStyles = makeStyles({
...
+
+  const handleClickOpenList = list => {
```
Comment (Esodey) [inline client/src/pages/Dashboard.js:23]: I can hopefully use this logic for the notifications/product details dialog as well

File: server/app/_data/seeder.js [hunk 7]
```diff
@@ -33,9 +35,11 @@ const users = JSON.parse(fs.readFileSync(`${__dirname}/users.json`, 'utf-8'));
 // Import data into database
 const importData = async () => {
   try {
-    await List.create(lists);
-    await Product.create(products);
-    await User.create(users);
+    await Promise.all([
```
Comment (Esodey) [inline server/app/_data/seeder.js:38]: What is the difference here when you add Promise.all?

**Review by Esodey — APPROVED** [review]
Looks great. Well done!

**Review by vic2019 — APPROVED** [review]

File: client/src/reducers/notifications.js [hunk 8]
```diff
@@ -12,6 +15,8 @@ export default function(state = initialState, action) {
         ...state,
         notifications: payload.notifications,
       };
+    case CLEAR_NOTIFICATIONS:
+      return initialState;
```
Comment (vic2019) [inline client/src/reducers/notifications.js:27]: This reminded me that I need to implement clean-up after user logout too..

**Pull request merged**